src/doc/src/faq.md

```diff
@@ -259,3 +259,49 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### What does "version conflict" mean and how to resolve it?
+
+> failed to select a version for `x` which could resolve this conflict
+Have you seen the error message above?
+
+This is one of the most annoying error message for Cargo users. There are several 
+situations may lead us to a version conflict. Below we'll walk through possible 
+causes and provide diagnostic techniques to help you out there:
+
+- The project and its dependencies use [links] to repeatedly link the local 
+  library. Cargo forbids linking two packages with the same native library, so 
+  even with multiple layers of dependencies it is not allowed. In this case, the 
+  error message will prompt: `Only one package in the dependency graph may specify 
+  the same links value`, you may need to manually check and delete duplicate link 
+  values. The community also have [conventions in place] to alleviate this.
+
+- When depending on different crates in the project, if these crates use the same 
+  dependent library, but the version used is restricted, making it impossible to 
+  determine the correct version, it will also cause conflicts. The error message 
+  will prompt: `all possible versions conflict with previously selected packages`. 
+  You may need to modify the version requirements to make them consistent.
+
+- If there are multiple versions of dependencies in the project, when using 
+  [`direct-minimal-versions`], the minimum version requirements cannot be met, 
+  which will cause conflicts. Please check in the dependency graph of the project, 
+  the version requirements for dependencies are inconsistent. Please modify the 
+  version requirements to make them consistent.
```

Comment:
This should be better understanding.

```suggestion
  which will cause conflicts. You may need to modify version requirements of your
  direct dependencies to meet the minimum SemVer version accordingly.
```

Reply:
Welcome to point out more questions, thanks：）